<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Jaehyun Joo" />


<title>A set-based association test in snpsettest</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>






<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">A set-based association test in snpsettest</h1>
<h4 class="author">Jaehyun Joo</h4>
<h4 class="date">17 January, 2022</h4>



<p>For set-based association tests, the <strong>snpsettest</strong> package employed the statistical model described in VEGAS (<strong>ve</strong>rsatile <strong>g</strong>ene-based <strong>a</strong>ssociation <strong>s</strong>tudy) [1], which takes as input variant-level p values and reference likage disequilibrium (LD) data. Briefly, the test statistics is defined as the sum of squared variant-level Z-statistics. Letting a set of <span class="math inline">\(Z\)</span> scores of individual SNPs <span class="math inline">\(z_i\)</span> for <span class="math inline">\(i \in 1:p\)</span> within a set <span class="math inline">\(s\)</span>, the test statistic <span class="math inline">\(Q_s\)</span> is</p>
<p><span class="math display">\[Q_s = \sum_{i=1}^p z_i^2\]</span></p>
<p>Here, <span class="math inline">\(Z = \{z_1,...,z_p\}&#39;\)</span> is a vector of multivariate normal distribution with a mean vector <span class="math inline">\(\mu\)</span> and a covariance matrix <span class="math inline">\(\Sigma\)</span> in which <span class="math inline">\(\Sigma\)</span> represents LD among SNPs. To test a set-level association, we need to evaluate the distribution of <span class="math inline">\(Q_s\)</span>. VEGAS uses Monte Carlo simulations to approximate the distribution of <span class="math inline">\(Q_s\)</span> (directly simulate <span class="math inline">\(Z\)</span> from multivariate normal distribution), and thus, compute a set-level p value. However, its use is hampered in practice when set-based p values are very small because the number of simulations required to obtain such p values is be very large. The <strong>snpsettest</strong> package utilizes a different approach to evaluate the distribution of <span class="math inline">\(Q_s\)</span> more efficiently.</p>
<p>Let <span class="math inline">\(Y = \Sigma^{-\frac12}Z\)</span> (instead of <span class="math inline">\(\Sigma^{-\frac12}\)</span>, we could use any decomposition that satisfies <span class="math inline">\(\Sigma = AA&#39;\)</span> with a <span class="math inline">\(p \times p\)</span> non-singular matrix <span class="math inline">\(A\)</span> such that <span class="math inline">\(Y = A^{-1}Z\)</span>). Then,</p>
<p><span class="math display">\[
\begin{gathered}
E(Y) = \Sigma^{-\frac12} \mu \\
Var(Y) = \Sigma^{-\frac12}\Sigma\Sigma^{-\frac12} = I_p \\
Y \sim N(\Sigma^{-\frac12} \mu,~I_p)
\end{gathered}
\]</span></p>
<p>Now, we posit <span class="math inline">\(U = \Sigma^{-\frac12}(Z - \mu)\)</span> so that</p>
<p><span class="math display">\[U \sim N(\mathbf{0}, I_p),~~U = Y - \Sigma^{-\frac12}\mu\]</span></p>
<p>and express the test statistic <span class="math inline">\(Q_s\)</span> as a quadratic form:</p>
<p><span class="math display">\[
\begin{aligned}
  Q_s &amp;= \sum_{i=1}^p z_i^2 = Z&#39;I_pZ = Y&#39;\Sigma^{\frac12}I_p\Sigma^{\frac12}Y \\
      &amp;= (U + \Sigma^{-\frac12}\mu)&#39;\Sigma(U + \Sigma^{-\frac12}\mu)
\end{aligned}
\]</span></p>
<p>With the spectral theorem, <span class="math inline">\(\Sigma\)</span> can be decomposed as follow:</p>
<p><span class="math display">\[
\begin{gathered}
\Sigma = P\Lambda P&#39; \\ \Lambda = \mathbf{diag}(\lambda_1,...,\lambda_p),~~P&#39;P
= PP&#39; = I_p
\end{gathered}
\]</span></p>
<p>where <span class="math inline">\(P\)</span> is an orthogonal matrix. If we set <span class="math inline">\(X = P&#39;U\)</span>, <span class="math inline">\(X\)</span> is a vector of independent standard normal variable <span class="math inline">\(X \sim N(\mathbf{0}, I_p)\)</span> since</p>
<p><span class="math display">\[E(X) = P&#39;E(U) = \mathbf{0},~~Var(X) = P&#39;Var(U)P = P&#39;I_pP = I_p\]</span></p>
<p><span class="math display">\[
\begin{aligned}
  Q_s &amp;= (U + \Sigma^{-\frac12}\mu)&#39;\Sigma(U + \Sigma^{-\frac12}\mu) \\
  &amp;= (U + \Sigma^{-\frac12}\mu)&#39;P\Lambda P&#39;(U + \Sigma^{-\frac12}\mu) \\
  &amp;= (X + P&#39;\Sigma^{-\frac12}\mu)&#39;\Lambda (X + P&#39;\Sigma^{-\frac12}\mu)
\end{aligned}
\]</span></p>
<p>Under the null hypothesis, <span class="math inline">\(\mu\)</span> is assumed to be <span class="math inline">\(\mathbf{0}\)</span>. Hence,</p>
<p><span class="math display">\[Q_s = X&#39;\Lambda X = \sum_{i=1}^p \lambda_i x_i^2\]</span></p>
<p>where <span class="math inline">\(X = \{x_1,...,x_p\}&#39;\)</span>. Thus, the null distribution of <span class="math inline">\(Q_s\)</span> is a linear combination of independent chi-square variables <span class="math inline">\(x_i^2 \sim \chi_{(1)}^2\)</span> (i.e., central quadratic form in independent normal variables). For computing a probability with a scalar <span class="math inline">\(q\)</span>,</p>
<p><span class="math display">\[Pr(Q_s &gt; q)\]</span></p>
<p>several methods have been proposed, such as numerical inversion of the characteristic function [2]. The <strong>snpsettest</strong> package uses the algorithm of Davies [3] or saddlepoint approximation [4] to obtain set-based p values.</p>
<p><strong>References</strong></p>
<ol style="list-style-type: decimal">
<li><p>Liu JZ, Mcrae AF, Nyholt DR, Medland SE, Wray NR, Brown KM, et al. A Versatile Gene-Based Test for Genome-wide Association Studies. Am J Hum Genet. 2010 Jul 9;87(1):139–45.</p></li>
<li><p>Duchesne P, De Micheaux P. Computing the distribution of quadratic forms: Further comparisons between the Liu-Tang-Zhang approximation and exact methods. Comput Stat Data Anal. 2010;54:858–62.</p></li>
<li><p>Davies RB. Algorithm AS 155: The Distribution of a Linear Combination of Chi-square Random Variables. J R Stat Soc Ser C Appl Stat. 1980;29(3):323–33.</p></li>
<li><p>Kuonen D. Saddlepoint Approximations for Distributions of Quadratic Forms in Normal Variables. Biometrika. 1999;86(4):929–35.</p></li>
</ol>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
