<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Jaehyun Joo" />


<title>Introduction to snpsettest</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to snpsettest</h1>
<h4 class="author">Jaehyun Joo</h4>
<h4 class="date">09 September, 2023</h4>



<p>This vignette shows you how to perform gene-based association tests using GWAS summary statistics in which sets of SNPs are defined by genes.</p>
<hr />
<div id="gwas-summary-statistics" class="section level2">
<h2>GWAS summary statistics</h2>
<p>The <strong>snpsettest</strong> requires SNP-level p-values to perform gene-based association tests.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(snpsettest)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="co"># Check an example of GWAS summary file (included in this package)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="kw">head</span>(exGWAS, <span class="dv">3</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a><span class="co">#&gt;      id chr   pos A1 A2    pvalue</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a><span class="co">#&gt; 1 SNP_0   1 50215  G  C 0.1969353</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a><span class="co">#&gt; 2 SNP_2   1 50768  A  G 0.6620465</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a><span class="co">#&gt; 3 SNP_3   1 50833  T  G 0.5822596</span></span></code></pre></div>
</div>
<div id="reference-data" class="section level2">
<h2>Reference data</h2>
<p>To infer the relationships among SNPs, the <strong>snpsettest</strong> package requires a reference data set. The GWAS genotype data itself can be used as the reference data (If the GWAS cohort is large, it is impractical to use genotype data of all individuals. It would be sufficient to randomly select 1,000 unrelated individuals for inferring pairwise LD correlations among common SNPs). Otherwise, you could use publicly available data, such as the 1000 Genomes (please see the companion vignette for <a href="reference_1000Genomes.html">processing the 1000 Genomes data</a>). This package accepts PLINK 1 binary files (.bed, .bim, .fam) as an input. We can use <code>read_reference_bed</code> to read them into R.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="co"># Path to .bed file</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a>bfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example.bed&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;snpsettest&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true"></a><span class="co"># Read a .bed file using bed.matrix-class in gaston package</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true"></a><span class="co"># Genotypes are retrieved on demand to manage large-scale genotype data</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true"></a>x &lt;-<span class="st"> </span><span class="kw">read_reference_bed</span>(bfile, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true"></a><span class="co">#&gt; Created a bed.matrix with 300 individuals and 2,942 markers.</span></span></code></pre></div>
</div>
<div id="harmonize-gwas-summary-to-the-reference-data" class="section level2">
<h2>Harmonize GWAS summary to the reference data</h2>
<p>Pre-processing of GWAS summary data is required because the sets of variants available in a particular GWAS might be poorly matched to the variants in reference data. SNP matching can be performed using <code>harmonize_sumstats</code> either 1) by SNP ID or 2) by chromosome code, base-pair position, and allele codes, while taking into account reference allele swap and possible strand flips.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="co"># Harmonize by SNP IDs</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a>hsumstats1 &lt;-<span class="st"> </span><span class="kw">harmonize_sumstats</span>(exGWAS, x)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a><span class="co">#&gt; Checking the reference data for harmonization...</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a><span class="co">#&gt; Found 0 monomoprhic SNPs in the reference data.</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true"></a><span class="co">#&gt; Found 0 duplicate SNP IDs in the reference data.</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true"></a><span class="co">#&gt; Excluded 0 SNPs from the harmonization.</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true"></a><span class="co">#&gt; Checking the GWAS summary statistics...</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true"></a><span class="co">#&gt; 2,753 variants to be matched.</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true"></a><span class="co">#&gt; 2,630 variants have been matched.</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true"></a><span class="co"># Harmonize by genomic position and allele codes</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true"></a><span class="co"># Reference allele swap will be taken into account (e.g., A/C match C/A)</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true"></a>hsumstats2 &lt;-<span class="st"> </span><span class="kw">harmonize_sumstats</span>(exGWAS, x, <span class="dt">match_by_id =</span> <span class="ot">FALSE</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true"></a><span class="co">#&gt; Checking the reference data for harmonization...</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true"></a><span class="co">#&gt; Found 0 monomoprhic SNPs in the reference data.</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true"></a><span class="co">#&gt; Found 0 duplicate SNPs in the reference data by genomic position and alleles codes.</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true"></a><span class="co">#&gt; Excluded 0 SNPs from the harmonization.</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true"></a><span class="co">#&gt; Checking the GWAS summary statistics...</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true"></a><span class="co">#&gt; 2,753 variants to be matched.</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true"></a><span class="co">#&gt; 2,618 variants have been matched.</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true"></a><span class="co"># Check matching entries by flipping allele codes (e.g., A/C match T/G)</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true"></a><span class="co"># Ambiguous SNPs will be excluded from harmonization</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true"></a>hsumstats3 &lt;-<span class="st"> </span><span class="kw">harmonize_sumstats</span>(exGWAS, x, <span class="dt">match_by_id =</span> <span class="ot">FALSE</span>, <span class="dt">check_strand_flip =</span> <span class="ot">TRUE</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true"></a><span class="co">#&gt; Checking the reference data for harmonization...</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true"></a><span class="co">#&gt; Found 0 monomoprhic SNPs in the reference data.</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true"></a><span class="co">#&gt; Found 0 duplicate SNPs in the reference data by genomic position and alleles codes.</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true"></a><span class="co">#&gt; Excluded 0 SNPs from the harmonization.</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true"></a><span class="co">#&gt; Checking the GWAS summary statistics...</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true"></a><span class="co">#&gt; 2,753 variants to be matched.</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true"></a><span class="co">#&gt; 835 ambiguous SNPs have been removed.</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true"></a><span class="co">#&gt; 1,795 variants have been matched.</span></span></code></pre></div>
</div>
<div id="map-snps-to-genes" class="section level2">
<h2>Map SNPs to genes</h2>
<p>To perform gene-based association tests, it is necessary to annotate SNPs onto their neighboring genes. Mapping SNPs to genes (or genomic regions) can be achieved by <code>map_snp_to_genes</code> with gene start/end information.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="co"># Check gene information from the GENCODE project (included in this package)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="kw">head</span>(gene.curated.GRCh37, <span class="dv">3</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a><span class="co">#&gt;             gene.id chr  start    end strand  gene.name      gene.type</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a><span class="co">#&gt; 1 ENSG00000186092.4   1  69091  70008      +      OR4F5 protein_coding</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a><span class="co">#&gt; 2 ENSG00000237683.5   1 134901 139379      - AL627309.1 protein_coding</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a><span class="co">#&gt; 3 ENSG00000235249.1   1 367640 368634      +     OR4F29 protein_coding</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a><span class="co"># Map SNPs to genes</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true"></a>snp_sets &lt;-<span class="st"> </span><span class="kw">map_snp_to_gene</span>(hsumstats1, gene.curated.GRCh37)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true"></a><span class="kw">str</span>(snp_sets<span class="op">$</span>sets[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true"></a><span class="co">#&gt; List of 5</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true"></a><span class="co">#&gt;  $ ENSG00000186092.4: chr [1:110] &quot;SNP_0&quot; &quot;SNP_2&quot; &quot;SNP_3&quot; &quot;SNP_4&quot; ...</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true"></a><span class="co">#&gt;  $ ENSG00000237683.5: chr [1:109] &quot;SNP_317&quot; &quot;SNP_320&quot; &quot;SNP_321&quot; &quot;SNP_323&quot; ...</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true"></a><span class="co">#&gt;  $ ENSG00000235249.1: chr [1:95] &quot;SNP_1283&quot; &quot;SNP_1285&quot; &quot;SNP_1287&quot; &quot;SNP_1288&quot; ...</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true"></a><span class="co">#&gt;  $ ENSG00000185097.2: chr [1:96] &quot;SNP_2392&quot; &quot;SNP_2396&quot; &quot;SNP_2397&quot; &quot;SNP_2398&quot; ...</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true"></a><span class="co">#&gt;  $ ENSG00000187634.6: chr [1:135] &quot;SNP_3455&quot; &quot;SNP_3456&quot; &quot;SNP_3458&quot; &quot;SNP_3459&quot; ...</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true"></a><span class="co"># Allows a certain kb window before/after the gene to be included for SNP mapping</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true"></a>snp_sets_50kb &lt;-<span class="st"> </span><span class="kw">map_snp_to_gene</span>(</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true"></a>  hsumstats1, gene.curated.GRCh37, </span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true"></a>  <span class="dt">extend_start =</span> <span class="dv">50</span>, <span class="dt">extend_end =</span> <span class="dv">50</span> <span class="co"># default is 20kb</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true"></a>)</span></code></pre></div>
</div>
<div id="perform-gene-based-association-tests" class="section level2">
<h2>Perform gene-based association tests</h2>
<p>Once we have SNP sets for genes, <code>snpset_test</code> can be used to perform gene-based association tests.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="co"># Perform gene-based association tests for the first 5 genes</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>res &lt;-<span class="st"> </span><span class="kw">snpset_test</span>(hsumstats1, x, snp_sets<span class="op">$</span>sets[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a><span class="co">#&gt; 2,630 variants are found in hsumstats1.</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a><span class="co">#&gt; 5 set-based association tests will be performed.</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a><span class="co">#&gt; Starting set-based association tests...</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a><span class="co">#&gt; -----</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a><span class="co">#&gt; ENSG00000186092.4: nSNP = 110, P = 0.042</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true"></a><span class="co">#&gt; ENSG00000237683.5: nSNP = 109, P = 0.00936</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true"></a><span class="co">#&gt; ENSG00000235249.1: nSNP = 95, P = 0.182</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true"></a><span class="co">#&gt; ENSG00000185097.2: nSNP = 96, P = 0.122</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true"></a><span class="co">#&gt; ENSG00000187634.6: nSNP = 135, P = 0.0103</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true"></a><span class="co"># Show output</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true"></a>res</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true"></a><span class="co">#&gt;               set.id    tstat      pvalue n.snp top.snp.id top.snp.pvalue</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true"></a><span class="co">#&gt; 1: ENSG00000186092.4 141.7800 0.042027775   110     SNP_78   0.0009143436</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true"></a><span class="co">#&gt; 2: ENSG00000237683.5 154.2858 0.009362739   109    SNP_363   0.0006419257</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true"></a><span class="co">#&gt; 3: ENSG00000235249.1 109.0270 0.182400780    95   SNP_1311   0.0047610286</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true"></a><span class="co">#&gt; 4: ENSG00000185097.2 114.7301 0.122042173    96   SNP_2458   0.0034444534</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true"></a><span class="co">#&gt; 5: ENSG00000187634.6 185.7576 0.010306441   135   SNP_3601   0.0003350840</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
